packages/core/src/exec.ts

```diff
@@ -157,8 +157,9 @@ function getActionFunction<TState extends AnyState>(
       const { actor } = (action as StopActionObject).params;
 
       if (actor) {
-        actor.stop?.();
-        delete state.children[actor.id];
```

Comment:
I'm not super confident about removing this `delete` but the current tests pass

Reply:
This might cause a memory leak

Reply:
In such a case, we should be able to verify this with a test - it's not a leak that could happen through closure or anything like that. If a reference to a child stays on the `state` then we can write an assertion for this thing